docs/src/main/sphinx/security/https.rst

```diff
@@ -0,0 +1,361 @@
+replacement.
+
+.. _cert-placement:
+
+Place the certificate file
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are no location requirements for a certificate file as long as:
+
+* The location is visible from the server's configuration file location with a
+  relative path reference from the server's root directory, or with an absolute
+  path reference.
+* The location is secure from copying or tampering by malicious actors.
+
+You can place your file in the Trino server's ``etc`` directory, which allows
+you to use a relative path reference in configuration files. However, this
+location can require you to keep track of the certificate file, and move it to
+a new ``etc`` directory when you upgrade your Trino version.
+
+.. _configure-https:
+
+Configure the coordinator
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+On the coordinator, add the following lines to the :ref:`config properties file
+<config_properties>` to enable HTTPS support for the server.
+
+.. note::
+
+  Legacy ``keystore`` and ``truststore`` wording is used in property names, even
+  when directly using PEM-encoded certificates. PEM certs generally do not
+  require truststore properties, because this format incorporates its chain of
+  authorizing certificates back to the originating CA (or possibly back to an
+  intermediate CA assigned by a large international CA).
+
+.. code-block:: text
+
+  http-server.https.enabled=true
+  http-server.https.port=8443
+  http-server.https.keystore.path=etc/clustercoord.pem
+
+Possible alternatives for the third line include:
+
+.. code-block:: text
+
+  http-server.https.keystore.path=etc/clustercoord.jks
+  http-server.https.keystore.path=/usr/local/certs/clustercoord.p12
+
+Relative paths are relative to the Trino server's root directory. In a
+``tar.gz`` installation, the root directory is one level above ``etc``.
+
+JKS keystores always require a password, while PEM format certificates can
+optionally require a password. For these cases, add the following line to
+specify the password.
+
+.. code-block:: text
+
+  http-server.https.keystore.key=<keystore-password>
+
+It is possible for a key inside a keystore to have its own password,
+independent of the keystore's password. In this case, specify the key's password
+with the following property:
+
+.. code-block:: text
+
+  http-server.https.keymanager.password=<key-password>
+
+When your Trino coordinator has an authenticator enabled along with HTTPS
+enabled, it disables HTTP access to the :doc:`Web UI </admin/web-interface>`.
+Although not recommended, you can re-enable it by setting:
+
+.. code-block:: text
+
+  http-server.authentication.allow-insecure-over-http=true
+
+When configured to provide HTTPS connections as shown above, the server
+continues to allow HTTP connections to all clients except the Web UI. When
+you are certain HTTPS connections are stable and reliable from the clients of
+interest, you can disable HTTP access:
```

Comment:
This is not true anymore.  Once  HTTPS is enabled with an authenticator, all client communication over HTTP is disabled.  HTTP is still used for inner cluster communication.  Switching that to HTTPS is covered in ther internal communications guide, and it a complex and difficult task, so not recommended.

Reply:
Anyway, I  would drop from this paragraph to the next section start.

Reply:
Well, OK, but I have a dim memory that someone in Support requested the addition of the `http-server.http.enabled=false` option to the docs. Let's let them ask again, if so.